core/src/main/java/io/grpc/ServerCall.java

```diff
@@ -176,4 +176,20 @@ public boolean isReady() {
   public void setMessageCompression(boolean enabled) {
     // noop
   }
+
+  /**
+   * Sets the compression algorithm for this call.  If the server does not support the compression
+   * algorithm, the call will fail.  This method may only be called once, and only before
```

Comment:
Why the limit of only being called once? Earlier that seemed to be an implementation detail, but now it is an explicit check for its own sake. Isn't it conceivable that one interceptor might set this and another override an earlier one? Yes, it could help detect a bug, but it could also make calling code less orthogonal and more coupled.